Client/Application/AppDelegate.swift

```diff
@@ -361,11 +361,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     /// Send a telemetry ping if the user hasn't disabled reporting.
     /// We still create and log the ping for non-release channels, but we don't submit it.
     private func sendCorePing() {
-        if let profile = profile where (profile.prefs.boolForKey("settings.sendUsageData") ?? true) {
-            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0)) {
-                let ping = CorePing(profile: profile)
-                Telemetry.sendPing(ping)
+        guard let profile = profile where (profile.prefs.boolForKey("settings.sendUsageData") ?? true) else {
+            log.debug("Usage sending is disabled. Not creating core telemetry ping.")
+            return
+        }
+
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0)) {
+            guard DeviceInfo.hasConnectivity() else {
+                log.debug("No connectivity. Not creating core telemetry ping.")
+                return
             }
+
+            let ping = CorePing(profile: profile)
```

Comment:
I'd explain why you don't try to send when you don't have connectivity – i.e. because it resets your measurements & increments seq, e.g. like your comment on this PR.